### PR TITLE
Make DiscreteResults Unchanging

### PR DESCRIPTION
**Pull request by jbrockmendel**
It is _really_ tough to figure out what is tested where, what has externally-confirmed tests vs internal-consistency tests vs smoketests.  This peels back one entirely unnecessarily layer of that.

### CODE REVIEW

**Comment by josef-pkt** [comment]
(found this open in a browser tab that I looked at)

I think main idea is good.
some renamings to make it closer to the usual Stata results files
Namespace -> Holder
obj -> res

and I think it's better without attributes, i.e. use method call which will load the relevant data in a lazy way, instead of attaching all results to the classes upon import.

**Comment by josef-pkt** [comment]
not urgent, maybe eventually

File: statsmodels/discrete/tests/test_count_model.py [hunk 1]
```diff
@@ -80,8 +80,7 @@ def setup_class(cls):
         cls.res1._results._attach_nullmodel = True
         cls.init_keys = ['exog_infl', 'exposure', 'inflation', 'offset']
         cls.init_kwds = {'inflation': 'logit'}
-        res2 = RandHIE()
-        res2.zero_inflated_poisson_logit()
+        res2 = RandHIE.zero_inflated_poisson_logit
```
Comment (bashtage) [inline statsmodels/discrete/tests/test_count_model.py]: These seem to change instances to functions.  Is this right?

Comment (jbrockmendel) [reply statsmodels/discrete/tests/test_count_model.py]: ATM `res2` is a  `RandHIE` instance, and when `zero_inflated_possion_logit` is called, all of `res2`'s relevant attributes are re-set.  After this PR, `zero_inflated_poisson_logit` is just an attribute of `RandHIE`, an object with the same attributes as `res2` in the test.  The PR is changing out a stateful object for a stateless one.

Comment (bashtage) [reply statsmodels/discrete/tests/test_count_model.py]: Oh -- github was hiding the large diff that changed the results file.  I was looking at the file in master and trying to figure out how this wasn't erroring.  This is "fine" then.

**Comment by bashtage** [comment]
Needs rebase, pls.

File: statsmodels/discrete/tests/test_count_model.py [hunk 2]
```diff
@@ -98,8 +97,7 @@ def setup_class(cls):
         cls.res1._results._attach_nullmodel = True
         cls.init_keys = ['exog_infl', 'exposure', 'inflation', 'offset']
         cls.init_kwds = {'inflation': 'probit'}
-        res2 = RandHIE()
-        res2.zero_inflated_poisson_probit()
+        res2 = RandHIE.zero_inflated_poisson_probit
```
Comment (bashtage) [inline statsmodels/discrete/tests/test_count_model.py]: Makes sense.

**Comment by bashtage** [comment]
lint failure.

**Comment by jbrockmendel** [comment]
azure fail is get_rdataset URLError

**Pull request merged**